### DISCUSSION TITLE
Welcome to OATS discussions!

### DISCUSSION THREAD

**Discussion by bukhsh** [bukhsh/oats#99] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other users of OATS. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other OATS users.
  * Welcome ideas and are open-minded about sharing data and extensions build using OATS. Remember that this is a tool that we build together 💪.

**Comment by bukhsh** [comment]
Hi, Many thanks for getting in touch. NEOS has recently made a requirement
for all users to submit an email address with their requests. Last time I
checked, PYOMO didn't have a variable to pass email id when calling NEOS
server. I have raised an issue and will sort it out as soon as an update
from pyomo is available. To answer your question about the demand modeling:
pD is the total demand met and 'alpha' is the proportion of the demand.
Alpha is a variable defined between 0 and 1. alpha=1 means that all demand
is met and alpha=0 means 0 demand is met. OATS model load shedding using
this decision variable. Hope this answers your question. Let me know if you
have any other questions. Best wishes, Waqquas

**Comment by Prashantgrid** [comment]
Dear Waqquas,

Thanks for the reply. I somehow managed to register my email id and now it
works perfectly.

I also wanted to ask you. How important is it to include the ramp rate
constraints of generators in a DC (as well as AC) opf?
I guess OATS also does not include the ramp rates.

Thanks and regards,
Prashant

ᐧ

**Reply by bukhsh** [reply]
Hi Prashant,

Apologies for the slow response. For some reason, I didn't get a notification of your question and I just stumbled upon this post. 

Ramp rates are not relevant for a single snap-shot model like ACOPF/DCOPF. They are used in time-coupled models like the unit commitment problem, or a multi-period DCOPF. Both of these models are available in the current release of OATS.

Waqquas

**Comment by clopezsalgado** [comment]
Is the code for the Convex AC power flow available?

**Comment by clopezsalgado** [comment]
I mean, the one discussed at:
https://strathprints.strath.ac.uk/87594/1/Bakhsh-2020-Convex-approximations-to-the-nonlinear-AC-Optimal.pdf
